user/util.go

```diff
@@ -0,0 +1,26 @@
+package user
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/address"
+)
+
+// IDFromPublicKey forms Neo wallet address from ecdsa.PublicKey and encodes it
+// into ID using buf. Buffer length MUST be at least 25 bytes (according to Neo3
+// docs).
+func IDFromPublicKey(id *ID, key ecdsa.PublicKey, buf []byte) {
+	scriptHash := (*keys.PublicKey)(&key).GetScriptHash()
+
+	buf[0] = address.Prefix
+	copy(buf[1:], scriptHash.BytesBE())
+	copy(buf[21:], hash.Checksum(buf[:21]))
+
+	err := id.Decode(buf)
+	if err != nil {
+		panic(fmt.Sprintf("unexpected error from ID.Decode: %v", err))
+	}
```

Comment:
`IDFromPublicKey` and `Decode` are in the same package, I think we can safely use a `copy` from `Decode` here. Also why do you use intermediate buffer? `id` points to the already allocated memory.

Reply:
Both suggestions require `ID` to provide the access to its buffer, do u think it is worth and safe?. I still think that we should deal with `ID` using exported methods even in the same package.

Reply:
I believe a package may use it's knowledge about the internal implementation to provide the fastest implementation.

Reply:
This will make the function bound to the internal implementation of the type, and it will be forced to change along with the type. In my opinion, this generates heavily maintainable code rather than speeds up anything.

Reply:
This is a very small function, so I wouldn't expect any of the future changes (which are themselves quite unlikely) to complicate things.

Reply:
I will take into account the arguments in similar cases in the future (it became outdated here).